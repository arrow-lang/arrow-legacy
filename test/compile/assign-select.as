let a;

if false {
  a = 34;
} else if true {
  if false {
    a = 30;
  } else {
    a = 30;
  }
} else {
  a = 50;
}

let b = a; // ok
