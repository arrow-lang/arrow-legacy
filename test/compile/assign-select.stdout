@value = internal global double 0.000000e+00

define internal void @assign-select..init() {
  br i1 false, label %1, label %2

; <label>:1                                       ; preds = %0
  br label %3

; <label>:2                                       ; preds = %0
  br label %3

; <label>:3                                       ; preds = %2, %1
  %4 = phi double [ 4.120000e+00, %1 ], [ 5.600000e+01, %2 ]
  store double %4, double* @value
  ret void
}

define i32 @main(i32, i8**, i8**) {
  call void @assign-select..init()
  ret i32 0
}
